### DISCUSSION TITLE
Unable to build when .parcelrc does not define extends

### DISCUSSION THREAD

**Discussion by lukyth** [parcel-bundler/parcel#4697] [General] [answered]
# 🐛 bug report

The build fails with the config and command below:

```sh
parcel build src/index.html --no-source-maps
```

## 🎛 Configuration (.babelrc, package.json, cli command)

`.parcelrc`
```json
{
  "namers": ["@parcel/namer-default"]
}

```

`babel.config.json`
```js
{
  "presets": [
    "@babel/preset-env",
    "@babel/preset-react",
    "@babel/preset-typescript"
  ],
  "plugins": [
    "@babel/plugin-transform-runtime",
    [
      "module-resolver",
      {
        "alias": {
          "@myorg": "../common/src"
        }
      }
    ]
  ]
}
```

But when I update `.parcelrc` to include `"extends": "@parcel/config-default"` as well, the build is successful.
```json
{
  "extends": "@parcel/config-default",
  "namers": ["@parcel/namer-default"]
}
```

## 🤔 Expected Behavior

The build should run successfully with `.parcelrc` without `extends`.

## 😯 Current Behavior

```
❯ npm run build

> @myorg/mypackage@0.0.0 build /Users/kanitkorn/Code/myorg/myrepo/packages/mypackage
> parcel build src/index.html --no-source-maps

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @myorg/mypackage@0.0.0 build: `parcel build src/index.html --no-source-maps`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @myorg/mypackage@0.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kanitkorn/.npm/_logs/2020-06-05T12_02_30_138Z-debug.log
```

## 💁 Possible Solution

I'm not sure if it's intended or not that `extends` is required in `.parcelrc`. If that's the case then it should be clear in the README.

If it's not intended, then the solution would be to make the build pass when `extends` is not define in `.parcelrc`.

## 🔦 Context

I was trying to create `.parcelrc` to specify some custom plugin. I started with some official plugin to see how it works, so I used the example I found in https://github.com/parcel-bundler/parcel#parcelrcnamers, but the build failed.
```json
{
  "namers": ["@parcel/namer-default"]
}
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | ^2.0.0-alpha.3.2
| Node             | 12.6.1
| npm/Yarn         | 6.13.4
| Operating System | macOS Catalina 10.15.5

**Comment by DeMoorJasper** [comment, marked as answer]
Extends is required if you want to extend the default config which is the case for you.

The following config you wanted to use would not do anything as it has no transformers, packagers, bundlers, ... configured.

```
{
  "namers": ["@parcel/namer-default"]
}
```

You can add all these things in the config if you want but in your case you probably just want a custom namer or something with the default behaviour for everything else. That's why the extends field exists to prevent you from having to go through all the trouble of creating your own config from scratch.

**Reply by danielpeter67** [reply]
please kindly explain is .parcelrc a folder or a built-in package or a package you install in npm.
i don't know what .parcelrc is

**Comment by lukyth** [comment]
I see. Thank you so much for your answer! 🙇 Does this mean if I want to add an extra reporter, I'd have to look up https://github.com/parcel-bundler/parcel/blob/v2/packages/configs/default/index.json and put every reporter there in my config, then add my extra reporter afterward?
```json
{
  "extends": "@parcel/config-default",
  "reporters": [
    "@parcel/reporter-cli",
    "@parcel/reporter-dev-server",
    "@parcel/reporter-bundle-analyzer",
    "@parcel/reporter-bundle-buddy",
    "parcel-reporter-extra-report"
  ]
}
```
I wonder, wouldn't this make it hard to keep the list in `reporters` in sync with `config-default`? Would it be easier for a user if Parcel take anything in the config as an addition on top of the default config?

**Reply by danielpeter67** [reply]
where can i unable extends in .parcelrc?
cause i am trying to use [parcel-plugin-static-files-copy] plugin

**Comment by mischnic** [comment]
> I wonder, wouldn't this make it hard to keep the list in reporters in sync with config-default?

Do this (works for nearly every plugin type):

```json
{
  "extends": "@parcel/config-default",
  "reporters": ["...", "parcel-reporter-extra-report"]
}
```

**Comment by lukyth** [comment]
I see! Thank you so much! I saw that this `"..."` hasn't been mentioned in the README yet. Should it be?

**Reply by mischnic** [reply]
It mentioned on https://parcel2-docs.now.sh/getting-started/plugin-config/ however

**Reply by lukyth** [reply]
Oh, sorry I missed that website 😓  Thank you so much :)

**Comment by zocky** [comment]
Chiming in almost a year later to say that .extends should default to "@parcel/config-default", since that will be by far the most common case.

**Reply by DavidNzube101** [reply]
;)
